Lib/test/test_zipfile/_path/test_path.py

```diff
@@ -665,7 +665,8 @@ def for_name(cls, name, archive):
 
         TODO: extract this functionality and re-use
         """
-        self = cls(filename=name, date_time=time.localtime(time.time())[:6])
+        get_time = os.environ.get('SOURCE_DATE_EPOCH') or time.time()
```

Comment:
I guess this does test the behavior, but this requires the person calling the tests to set the environment variable -- that won't always happen.

Could you add a *new* test that explicitly sets the variable?